Tasks/PerformActionOnMachineGroup/PerformActionOnMachineGroup.ps1

```diff
@@ -27,7 +28,15 @@ Initialize-DTLServiceHelper
 
 $machineGroup = Get-MachineGroup -machineGroupName $MachineGroupName -filters $Filters
 
-$providerName = $machineGroup.Provider.Name
+# if providerName is null or empty then follow same path as standard environment.
+if([string]::IsNullOrEmpty($machineGroup.Provider) -eq $true)
```

Comment:
this will not be string type, isn't it? how would String method apply on provider?